compiler/utils/astrepr.nim

```diff
@@ -122,124 +200,175 @@ proc format[I](s: set[I], sub: int): string =
 
   result.add "}"
 
-const setStyle = termFg(2, 1, 3)
+func format(i: SomeInteger): ColText = $i + style.number
+
+proc ownerChain(sym: PSym): string =
+  var own: seq[PSym] = @[sym]
+  var sym = sym
+  while not sym.owner.isNil():
+    sym = sym.owner
+    own.add sym
 
-func format(i: SomeInteger): ColText = $i + fgCyan
+  for idx in countdown(own.high, 0):
+    if idx < own.high:
+      result.add "."
+
+    result.add own[idx].name.s
+    result.add "("
+    result.add substr($own[idx].kind, 2)
+    result.add ")"
+
+func defaulted(r: TReprConf): bool = trfShowDefaultedFields in r
 
 proc symFields(
     conf: ConfigRef,
     sym: PSym,
-    flags: set[TReprFlag] = defaultTreeReprFlags,
+    rconf: TReprConf = implicitTReprConf,
     indent: int = 0
   ): ColText =
 
   coloredResult(1)
   var res = addr result
-  genFields(res[], indent, flags)
+  genFields(res[], indent, rconf)
 
-  if trfShowSymFlags in flags and sym.flags.len > 0:
-    field("flags", format(sym.flags, 2) + setStyle)
+  if trfShowSymFlags in rconf and
```

Comment:
Don't have to do this now, but could leave a small refactor comment for others to follow up on, something like:

1. Create a const with sym related trfReprConf set, if it doesn't exist
2. Change if/Elif to a case statement on the kind
3. Embed the remaining conditionals within each statement branch
4. In the else guard to make sure no kind in the const is missed in prior conditions (other side of exhaustiveness)

Reply:
I don't understand, what do you want to change here. I have a series of conditionals, each check whether the field is enabled, and if so, they add it. Wrapping this all in case statement will only needlessly increase the nesting.